go/mcap/mcap.go

```diff
@@ -214,12 +216,65 @@ type ChunkIndex struct {
 // calibration data, or other arbitrary data. Attachment records must not appear
 // within a chunk.
 type Attachment struct {
-	LogTime    uint64
-	CreateTime uint64
-	Name       string
-	MediaType  string
-	Data       []byte
-	CRC        uint32
+	LogTime       uint64
+	CreateTime    uint64
+	Name          string
+	MediaType     string
+	ContentLength uint64
+	Data          io.Reader
+}
+
+// AttachmentReader represents an attachment for handling in a streaming manner.
+type AttachmentReader struct {
+	LogTime       uint64
+	CreateTime    uint64
+	Name          string
+	MediaType     string
+	ContentLength uint64
+	Data          io.Reader
+
+	baseReader io.Reader
+	crcReader  *crcReader
+	crc        *uint32
+}
+
+// ComputedCRC discards any remaining data in the Data portion of the
+// AttachmentReader, then returns the checksum computed from the fields of the
+// attachment up to the CRC. If it is called before the data portion of the
+// reader has been fully consumed, an error will be returned.
+func (ar *AttachmentReader) ComputedCRC() (uint32, error) {
+	n, err := io.Copy(io.Discard, ar.Data)
```

Comment:
this should just read one byte - that's all that's necessary to check if we need to fail